### PR TITLE
More setup and update readme

### PR DESCRIPTION
**Pull request by CatalinVoineag**
## What?
- After generating a new app with the rails template a few changes were needed, db setup, update gems, etc.

## Change considerations
- Added other things like `factory-bot, brakeman, view_components, etc`
- I might have missed a few things when generating the app so please feel free to add any comments and suggestions.
- People from placement, do you mind adding some description for your service as a comment to this PR? If not we can leave it blank.
- Please also have a look at the [initial commit](https://github.com/DFE-Digital/itt-mentor-services/commit/f7b293c78f0480eaf00d4c84c807dcc2768292bf)

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,47 +1,50 @@
...
 
 # Reduces boot times through caching; required in config/boot.rb
-gem "bootsnap", require: false
+gem 'bootsnap', require: false
 
 # Use Active Storage variants [https://guides.rubyonrails.org/active_storage_overview.html#transforming-images]
-# gem "image_processing", "~> 1.2"
+# gem 'image_processing', '~> 1.2'
 
 group :development, :test do
```
Comment (JamieCleare2525) [inline Gemfile:45]: Could you move this down to the `group :test, :development` grouping? Line:80
It seems odd for it to be isolated here.

Comment (CatalinVoineag) [reply Gemfile:45]: Yeah, good shout

**Review by ollietreend — COMMENTED** [review]
I can see a bunch of these quotation marks have changed from double (`"`) to single (`'`) quotes.

Was that done automatically by rubocop?

It looks like some of them are still double quotes (e.g. files in the `spec/` directory). I'm wondering if that's an inconsistency introduced by us, or by rubocop 🤔

**Comment by CatalinVoineag** [comment]
> I can see a bunch of these quotation marks have changed from double (`"`) to single (`'`) quotes.
> 
> Was that done automatically by rubocop?
> 
> It looks like some of them are still double quotes (e.g. files in the `spec/` directory). I'm wondering if that's an inconsistency introduced by us, or by rubocop 🤔

It's us. I don't think rubocop is looking at quotes. When it comes to the gemfile. The rails template uses double quotes but the `apply_for_teacher_training` gem file is using single, so I thought people prefer single quotes.

That said, I forgot to run bin/lint, which will make robocop change some things but not the double quotes

**Review by JamieCleare2525 — APPROVED** [review]
LGTM

**Comment by CatalinVoineag** [comment]
The linting issue has been fixed in this PR https://github.com/DFE-Digital/itt-mentor-services/pull/3#pullrequestreview-1765201287

**Pull request merged**